src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/needbraces/InputNeedBracesPatternMatchingForSwitchAllowSingleLine.java

```diff
@@ -84,10 +84,11 @@ case Point(int x, int y ) when (x>=0 && y >=0) :  // violation
                 yield 0;
         };
 
-        int b =  switch (o) {
+        int b;
+        b =  switch (o) {
```

Comment:
This change is for code coverage line 439 (  `|| switchAST.getParent().getParent().getType() == TokenTypes.EXPR;`)

Reply:
What if we have doubly nested switch expressions? Please add this to the new test cases.

Reply:
done added to the new input